priv/perf/apps/load_test/lib/service/faucet.ex

```diff
@@ -45,48 +69,124 @@ defmodule LoadTest.Service.Faucet do
   end
 
   def init(config) do
-    fee_wei = Keyword.fetch!(config, :fee_wei)
+    {:ok, faucet_account} = Account.new(Keyword.fetch!(config, :faucet_private_key))
+    Logger.debug("Using faucet: #{Encoding.to_hex(faucet_account.addr)}")
 
-    {faucet, eth_utxo} = get_funded_faucet_account(config)
-    Logger.debug("Using faucet: #{Encoding.to_hex(faucet.addr)}")
+    state =
+      struct!(
+        __MODULE__,
+        faucet_account: faucet_account,
+        fee: Keyword.fetch!(config, :fee_wei)
+      )
 
-    state = struct!(__MODULE__, account: faucet, fee: fee_wei, utxos: %{@eth => eth_utxo})
     {:ok, state}
   end
 
-  def handle_call(:get_faucet, _from, %__MODULE__{account: faucet} = state) do
-    {:reply, {:ok, faucet}, state}
+  def handle_call(:get_faucet, _from, %__MODULE__{faucet_account: faucet_account} = state) do
+    {:reply, {:ok, faucet_account}, state}
   end
 
-  def handle_call({:fund_child_chain, account, amount, @eth = token}, _from, %__MODULE__{account: faucet} = state) do
-    {utxo, utxo_amount} = state.utxos[token]
-    change = utxo_amount - amount - state.fee
-    if change < 0, do: raise({:error, :change_below_zero})
+  def handle_call(
+        {:fund_child_chain, receiver, amount, currency},
+        _from,
+        %__MODULE__{
+          faucet_account: faucet_account
+        } = state
+      ) do
+    utxo = check_sufficient_funds(state, currency, amount)
 
-    Logger.debug("Funding user: #{Encoding.to_hex(account.addr)} with UTXO: #{Utxo.pos(utxo)}")
+    change = utxo.amount - amount - state.fee
+    if change < 0, do: raise({:error, :insufficient_faucet_funds})
 
     outputs = [
-      %Utxo{amount: change, currency: token, owner: faucet.addr},
-      %Utxo{amount: amount, currency: token, owner: account.addr}
+      %Utxo{amount: change, currency: currency, owner: faucet_account.addr},
+      %Utxo{amount: amount, currency: currency, owner: receiver.addr}
     ]
 
-    {:ok, blknum, txindex} = Transaction.submit_tx([utxo], outputs, [faucet], @fund_child_chain_account_retries)
-    {:ok, change_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 0})
-    {:ok, user_utxo} = Utxo.new(%{blknum: blknum, txindex: txindex, oindex: 1, amount: amount})
+    Logger.debug("Funding user #{Encoding.to_hex(receiver.addr)} with #{amount} from utxo: #{Utxo.pos(utxo)}")
```

Comment:
I think we usually do 
`_ = Logger.debug(...`

Reply:
that would be necessary if this project used dialyzer

Reply:
this is sth I am pretty confused too? what is the reason/benefit for that extra `_`??

Reply:
in elixir and erlang every statement (line of code) is returning a value. there are no `void` functions. because of that, elixir and erlang are very expressive languages and because of that - it's considered an error of sorts, if you do something, but don't check/use what an invocation is returning.

so when you code elixir, you're supposed to `code for the correct case`. that means that if your functions correct execution produces `:ok` you would pattern match on `:ok`. 
`:ok = is_utxo_valid(utxo)` and if it's not, you'd let the supervision tree handle the exception. that of course is not always desirable, so you can go 
case is_utxo_valid(utxo)
:ok -> do this
_ -> do something else
end
etc.

so back to the initial question `Logger.debug/1` is the same thing, it invokes a function in a module, but because we're pattern matching into `_`, we're saying, `we don't care what it returns`.